### PR TITLE
Add title to profile pages (and fix a couple of things per code review)

### PR DESCRIPTION
**Pull request by yipingliao**
@2is10 
- Add title to user profile pages
- Fix hardcoded route to profile page on owner image in profile library card
- Refactor `userService.getProfileUrl` to be `routeService.getProfileUrl`

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js [hunk 1]
```diff
@@ -64,12 +64,17 @@ angular.module('kifi')
...
       userProfileActionService.getProfile(username).then(function (profile) {
+        setTitle(profile);
         initProfile(profile);
         initViewingUserStatus();
         initUserNavLinks();
       });
     }
 
+    function setTitle(profile) {
+      $window.document.title = profile.firstName + ' ' + profile.lastName + ' • Kifi' ;
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:75]: This is where the title is set. It's kinda deeply buried among all the refactoring name changes.

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:75]: I’d prefer that we configure page titles where we configure our routes, so it’s easy to see review them all in one place, to help ensure consistency, and to help remind people to set a title when adding a new route.

Here are some ways to do it with ui-router that a quick search turned up:
http://stackoverflow.com/questions/23813599/set-page-title-using-ui-router
https://coderwall.com/p/zk7qww/dynamic-site-page-titles-with-angular-and-ui-router

**Comment by 2is10** [comment]
:+1: Thanks!

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:75]: Hmm... I spent some time on this and haven't been able to find a more elegant way - reason being the title we want to set depends on the user's first and last names, which needs a backend call and is not yet available at the time that the routing is done.

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:75]: Thanks for trying. I’ll think about this along with delayed (promised-based) navigation over the next few days.

**Pull request merged**